src/snapred/backend/data/DataFactoryService.py

```diff
@@ -27,7 +29,15 @@ def defaultClass(self, val, clazz):
             val = clazz()
         return val
 
-    def getReductionIngredients(self, runId: str, pixelGroup: PixelGroup) -> ReductionIngredients:
+    def getReductionIngredients(self, runId: str, pixelGroup: PixelGroup = None) -> ReductionIngredients:
+        if pixelGroup is None:
+            calibration = self.getCalibrationState(runId)  # noqa: F811
+            if calibration is None or calibration.instrumentState.pixelGroup is None:
+                stateId = self.constructStateId(runId)
+                raise RuntimeError(f"Pixel group not found for runId {runId} and stateId {stateId}")
+
+            pixelGroup = calibration.instrumentState.pixelGroup
+
```

Comment:
This looks like an especially implemented method for this interface

Reply:
its not touching disk directly so its probably fine?